lib-rpc/ocamlformat_rpc_lib.ml

```diff
@@ -0,0 +1,197 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                              OCamlFormat                               *)
+(*                                                                        *)
+(*            Copyright (c) Facebook, Inc. and its affiliates.            *)
+(*                                                                        *)
+(*      This source code is licensed under the MIT license found in       *)
+(*      the LICENSE file in the root directory of this source tree.       *)
+(*                                                                        *)
+(**************************************************************************)
+
+open Sexplib0
+
+module type Command_S = sig
+  type t
+
+  val read_input : Stdlib.in_channel -> t
+
+  val to_sexp : t -> Sexp.t
+
+  val output : Stdlib.out_channel -> t -> unit
+end
+
+module type Client_S = sig
+  type t
+
+  type cmd
+
+  val pid : t -> int
+
+  val mk : pid:int -> in_channel -> out_channel -> t
+
+  val query : cmd -> t -> cmd
+
+  val halt : t -> (unit, [`Msg of string]) result
```

Comment:
```suggestion
  val halt : t -> (unit, [> `Msg of string]) result
```
I would suggest using open sets of variant in the Client's signature.
It allows for more flexible handling of `Result` mapping.

Reply:
Good idea, done!